### PR TITLE
No rx at all

### PR DESCRIPTION
**Pull request by hannahhoward**
This removes RxJS entirely, relying on angular 1.x's built in events on $rootScope to provide the underlying service for the EventEmitter class.

This code is pretty hacky, but the resulting minified file is only 2kb larger than the original version with RxJS not included.

### CODE REVIEW

**Comment by hannahhoward** [comment]
I think this is pretty hacky as is and I'd want to clean it up -- just want to know if you think it's worth pursuing this thread.

File: src/util/event-emitter.js [hunk 1]
```diff
@@ -1,32 +1,38 @@
...
-// Simple re-implementation of Angular 2's [EventEmitter](https://github.com/angular/angular/blob/master/modules/angular2/src/core/facade/async.ts#L137)
-import Subject from '@reactivex/rxjs/dist/cjs/Subject';
+// Simple re-implementation of Angular 2's [EventEmitter](https://github.com/angular/angular/blob/master/modules/angular2/src/facade/async.ts#L97)
+import eventDispatcher from "./event-dispatcher.js";
+
+const uuid = function() {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+    var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
+    return v.toString(16);
+  });
```
Comment (petebacondarwin) [inline src/util/event-emitter.js]: Is this really the uuid generator that ng2 is using?

**Comment by MikeRyan52** [comment]
What do you think it would take to get this code to a point where it attempts to use Rx and falls back onto our custom event emitter?

File: src/util/event-emitter.js (hunk 1 above)
Comment (hannahhoward) [inline src/util/event-emitter.js]: no like I said this was some hack-a-day stuff. I just pulled a uuid generator off stack overflow.

**Comment by hannahhoward** [comment]
@MikeRyan52 I guess the trick would be how to determine if RX is present? I should note I think my original estimate of the file with the new version of RxJS is only I think 9k larger when minified -- the total file size is probably around 40k with the right minification options on. So I wonder if this PR is even worth it. RxJS is battle tested... this approach, not so much :)

**Comment by timkindberg** [comment]
Looks like they add Rx to the global/window namespace: https://github.com/Reactive-Extensions/RxJS/blob/master/dist/rx.all.js#L12222-L12238

**Comment by timkindberg** [comment]
I think this one is null/void with the way we build the new typescript branch.

**Pull request closed without merging**